framework/src/fvbcs/FVMatAdvectionOutflowBC.C

```diff
@@ -23,23 +23,58 @@ FVMatAdvectionOutflowBC::validParams()
       "is acting on");
   params.addClassDescription(
       "Outflow boundary condition taking the advected quantity from a material property");
+  MooseEnum advected_interp_method("average upwind", "upwind");
+
+  params.addParam<MooseEnum>("advected_interp_method",
+                             advected_interp_method,
+                             "The interpolation to use for the advected quantity. Options are "
+                             "'upwind' and 'average', with the default being 'upwind'.");
   return params;
 }
 
 FVMatAdvectionOutflowBC::FVMatAdvectionOutflowBC(const InputParameters & params)
   : FVFluxBC(params),
-    _vel(getADMaterialProperty<RealVectorValue>("vel")),
-    _adv_quant(isParamValid("advected_quantity")
-                   ? &getADMaterialProperty<Real>("advected_quantity").get()
-                   : &_u)
+    _vel_elem(getADMaterialProperty<RealVectorValue>("vel")),
+    _vel_neighbor(getNeighborADMaterialProperty<RealVectorValue>("vel")),
+    _adv_quant_elem(isParamValid("advected_quantity")
+                        ? getADMaterialProperty<Real>("advected_quantity").get()
+                        : _u),
+    _adv_quant_neighbor(isParamValid("advected_quantity")
+                            ? getNeighborADMaterialProperty<Real>("advected_quantity").get()
+                            : _u_neighbor)
 {
+  using namespace Moose::FV;
+
+  const auto & advected_interp_method = getParam<MooseEnum>("advected_interp_method");
+  if (advected_interp_method == "average")
+    _advected_interp_method = InterpMethod::Average;
+  else if (advected_interp_method == "upwind")
+    _advected_interp_method = InterpMethod::Upwind;
+  else
+    mooseError("Unrecognized interpolation type ",
+               static_cast<std::string>(advected_interp_method));
 }
 
 ADReal
 FVMatAdvectionOutflowBC::computeQpResidual()
 {
-  mooseAssert(_normal * _vel[_qp] >= 0,
+  ADRealVectorValue v;
+  ADReal adv_quant_boundary;
+
+  using namespace Moose::FV;
+
+  // Currently only Average is supported for the velocity
+  interpolate(InterpMethod::Average, v, _vel_elem[_qp], _vel_neighbor[_qp], *_face_info, true);
+
+  interpolate(_advected_interp_method,
```

Comment:
with upwind interpolation this will disagree with the advection kernel for dirichlet BCs on advected_quantity. 
Not sure what the solution is though. Maybe just an if statement?

Reply:
How is that? The `computeQpResidual` here mirrors `FVMatAdvection::computeQpResidual`

Reply:
oh true. I guess I assumed that was not executed on dirichlet BCs

Reply:
It's a little funny...on boundaries with FVDirichletBCs we execute FVFluxKernels in order to weakly achieve the dirichlet condition. However, what you're poking around at here is that: if you have a Dirichlet boundary condition on an outflow boundary for a pure advection problem with an upwind interpolation scheme for the advected quantity, *then* your simulation will actually not feel that Dirichlet information at all. I believe it was you actually who revealed this in a Pronghorn simulation in which you were upwinding the pressure and consequently your result wasn't obeying the Dirichlet condition you had applied.

Reply:
ok resolved in FV meeting. I have no quarrel with upwinding ignoring downwind BCs, and for upwind BCs, FVFluxBCs are the call.